Shared/Store/BaseDataStore.swift

```diff
@@ -269,19 +262,44 @@ extension BaseDataStore {
 
 // locking management
 extension BaseDataStore {
-    public func unlock() {
+    private func setupAutolock() {
+        let emitsOnNewLifecycleObservable = Observable.combineLatest(self.lifecycleStore.lifecycleEvents, self.storageState)
+                .distinctUntilChanged { (lhLifecycleState, rhLifecycleState) -> Bool in
+                    return lhLifecycleState.0 == rhLifecycleState.0
+                }
```

Comment:
Squinting, I wander if this could be better done with a 

```swift
self.lifecycleStore.lifecycleEvents
    .distinctUntilChanged()
    .switchMap {}
```

but I'm not completely sure.

Reply:
No, using `combineLatest` means that the same version of the lifecycle will get pushed thru with each subsequent state, which isn't desirable. `zip` isn't quite right either; we want the most updated state variable when getting a new lifecycle action.

Reply:
oh wait, I see what you're saying